### PR TITLE
Add Email Notifier To Notification Hook

### PR DESCRIPTION
**Pull request by fuhrmeistery**
- Add Implementation For Email Notifier
- Add Basic Email Template
- Add Basic Test For Email Notifier

### CODE REVIEW

File: hooks/notification-hook/values.yaml [hunk 1]
```diff
@@ -48,5 +48,9 @@ env:
       secretKeyRef:
         name: some-secret
         key: some-key
-  - name: key
-    value: value
+  - name: SMTP_CONFIG
+    valueFRom:
```
Comment (J12934) [inline hooks/notification-hook/values.yaml:52]:
```suggestion
    valueFrom:
```

File: hooks/notification-hook/Notifiers/EMailNotifier.test.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      name: "demo-scan-1601086432",
+      namespace: "my-scans",
+      creationTimestamp: new Date("2021-01-01T14:29:25Z"),
+      labels: {
+        company: "iteratec",
+        "attack-surface": "external",
+      },
+    },
+    spec: {
+      scanType: "Nmap",
```
Comment (J12934) [inline hooks/notification-hook/Notifiers/EMailNotifier.test.ts:52]:
```suggestion
      scanType: "nmap",
```

File: hooks/notification-hook/README.md [hunk 3]
```diff
@@ -98,6 +103,36 @@ To configure a slack notification set the `type` to `slack` and the `endPoint` t
...
+This config reflects the transporter configuration of nodemailer (See [nodemailer | SMTP Transport](https://nodemailer.com/smtp/)).
+This configuration needs to be specified under `env` in the values yaml.
+The identifier for this config has to be `SMTP_CONFIG`.
+A basic configuration could look like this:
+
+```
+...
+env:
+  - name: SMTP_CONFIG
+    value: "smtp://kaden.lueilwitz48@ethereal.email:pezuHHseu8sS2QGcRS@smtp.ethereal.email/"
```
Comment (rseedorff) [inline hooks/notification-hook/README.md:122]: The example looks a bit ... strange 🤔
```suggestion
    value: "smtp://youraccount.yourpassword@ethereal.email:FOOBARXYZ@smtp.ethereal.email/"
```

File: hooks/notification-hook/README.md [hunk 4]
```diff
@@ -98,6 +103,36 @@ To configure a slack notification set the `type` to `slack` and the `endPoint` t
...
+```
+
+To provide a custom `from` field for your email you can specify `EMAIL_FROM` under env.
+For example:
+
+```
+env:
+  - name: SMTP_CONFIG
+    value: "smtp://kaden.lueilwitz48@ethereal.email:pezuHHseu8sS2QGcRS@smtp.ethereal.email/"
```
Comment (rseedorff) [inline hooks/notification-hook/README.md:131]:
```suggestion
    value: "smtp://youraccount.yourpassword@ethereal.email:FOOBARXYZ@smtp.ethereal.email/""
```

File: hooks/notification-hook/README.md.gotmpl [hunk 5]
```diff
@@ -98,6 +103,37 @@ To configure a slack notification set the `type` to `slack` and the `endPoint` t
...
+This config reflects the transporter configuration of nodemailer (See [nodemailer | SMTP Transport](https://nodemailer.com/smtp/)).
+This configuration needs to be specified under `env` in the values yaml.
+The identifier for this config has to be `SMTP_CONFIG`.
+A basic configuration could look like this:
+
+```
+...
+env:
+  - name: SMTP_CONFIG
+    value: "smtp://kaden.lueilwitz48@ethereal.email:pezuHHseu8sS2QGcRS@smtp.ethereal.email/"
```
Comment (rseedorff) [inline hooks/notification-hook/README.md.gotmpl:122]: see comment above

File: hooks/notification-hook/README.md.gotmpl [hunk 6]
```diff
@@ -98,6 +103,37 @@ To configure a slack notification set the `type` to `slack` and the `endPoint` t
...
+```
+
+To provide a custom `from` field for your email you can specify `EMAIL_FROM` under env.
+For example:
+
+```
+env:
+  - name: SMTP_CONFIG
+    value: "smtp://kaden.lueilwitz48@ethereal.email:pezuHHseu8sS2QGcRS@smtp.ethereal.email/"
```
Comment (rseedorff) [inline hooks/notification-hook/README.md.gotmpl:131]: see comment above

**Review by rseedorff — CHANGES_REQUESTED** [review]
Just some questions and minor improvements...

File: hooks/notification-hook/README.md (hunk 3 above)
Comment (fuhrmeistery) [reply hooks/notification-hook/README.md:122]: According to the documentation of nodemailer this should be a general connection URL for SMTP users and not just etheral.
https://nodemailer.com/smtp/

**Review by rseedorff — APPROVED** [review]

**Pull request merged**